components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -724,20 +729,23 @@ private void createFile(URI parentName, String newFileName) throws Exception {
      *
      * @param parentName
      * @param newFileName
-     * @throws Exception
+     * @param metadata
+     * @throws SynapseException
      */
-    private void writeToFile(URI parentName, String newFileName, String content) throws Exception {
+    private void writeToFile(URI parentName, String newFileName, String content, Properties metadata) throws SynapseException {
```

Comment:
`SynapseException` is a RuntimeException. Therefore we do not need a  throws declaration.